### PR TITLE
COSMA: Support MPItrampoline

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: C/COSMA/build_tarballs.jl [hunk 1]
```diff
@@ -53,37 +70,63 @@ cmake ../cosma \
...
+platforms = filter(p -> !(p["mpi"] == "openmpi" && arch(p) == "armv6l" && libc(p) == "glibc"), platforms)
+# MPItrampoline
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && (Sys.iswindows(p) || libc(p) == "musl")), platforms)
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && Sys.isfreebsd(p)), platforms)
+append!(dependencies, platform_dependencies)
+
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"7.1.0")
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
+               augment_platform_block, julia_compat="1.6", preferred_gcc_version = v"7.1.0")
```
Comment (giordano) [inline C/COSMA/build_tarballs.jl:132]: If you assume that OpenBLAS_jll on aarch64 does ILP64, then you must require Julia v1.7

Comment (eschnett) [reply C/COSMA/build_tarballs.jl:132]: Yes it does. Will do.

File: C/COSMA/build_tarballs.jl [hunk 2]
```diff
@@ -53,37 +71,61 @@ cmake ../cosma \
...
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = supported_platforms()
 platforms = expand_cxxstring_abis(platforms)
+
+# We need to expand the gfortran versions because MPItrampoline
+# depends on Fortran.
+platforms = expand_gfortran_versions(platforms)
```
Comment (giordano) [inline C/COSMA/build_tarballs.jl:100]: Same comment as https://github.com/JuliaPackaging/Yggdrasil/pull/5140#discussion_r919457600

Comment (eschnett) [reply C/COSMA/build_tarballs.jl]: Done.

**Pull request merged**